### PR TITLE
add test project filter

### LINKED ISSUE #43: Add support for filtering test projects in solution files
When specifying a `.sln` file, which usually includes production and tests projects, it should be possible to pass a `--include <regex>` or `--exclude <regex>` to filter out .Tests.csproj files from the solution.

### PR DESCRIPTION
**Pull request by Bertk**
check  `IsTestProject` in csproj  (resolves #43)
```
    <IsTestProject>true</IsTestProject>
```

### CODE REVIEW

**Review by coderpatros — CHANGES_REQUESTED** [review]
Hi @Bertk, looks pretty good. There's just a few tweaks I'd like.

File: CycloneDX.Core/Services/IProjectFileService.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ namespace CycloneDX.Services
 {
     public interface IProjectFileService
     {
-        Task<HashSet<NugetPackage>> GetProjectNugetPackagesAsync(string projectFilePath);
-        Task<HashSet<NugetPackage>> RecursivelyGetProjectNugetPackagesAsync(string projectFilePath);
+        Task<HashSet<NugetPackage>> GetProjectNugetPackagesAsync(string projectFilePath, bool excludeDev);
+        Task<HashSet<NugetPackage>> RecursivelyGetProjectNugetPackagesAsync(string projectFilePath, bool excludeDev);
         Task<HashSet<string>> GetProjectReferencesAsync(string projectFilePath);
         Task<HashSet<string>> RecursivelyGetProjectReferencesAsync(string projectFilePath);
     }
```
[the remark this answers is not in the document]
Comment (Bertk) [reply CycloneDX.Core/Services/IProjectFileService.cs]: Done.

File: CycloneDX/Program.cs [hunk 2]
```diff
@@ -167,15 +167,15 @@ public static async Task<int> Main(string[] args)
...
                 else if (Core.Utils.IsSupportedProjectType(SolutionOrProjectFile) && scanProjectReferences)
                 {
-                    packages = await projectFileService.RecursivelyGetProjectNugetPackagesAsync(fullSolutionOrProjectFilePath).ConfigureAwait(false);
+                    packages = await projectFileService.RecursivelyGetProjectNugetPackagesAsync(fullSolutionOrProjectFilePath, excludeDev).ConfigureAwait(false);
                 }
                 else if (Core.Utils.IsSupportedProjectType(SolutionOrProjectFile))
                 {
-                    packages = await projectFileService.GetProjectNugetPackagesAsync(fullSolutionOrProjectFilePath).ConfigureAwait(false);
+                    packages = await projectFileService.GetProjectNugetPackagesAsync(fullSolutionOrProjectFilePath, excludeDev).ConfigureAwait(false);
                 }
```
[the remark this answers is not in the document]
Comment (Bertk) [reply CycloneDX/Program.cs]: Done

**Comment by Bertk** [comment]
> Hi @Bertk, looks pretty good. There's just a few tweaks I'd like.

All tweaks are implemented.

**Comment by coderpatros** [comment]
Thanks @Bertk I'll try to get your PRs merged, and a new release done, over the next few days

**Review by coderpatros — APPROVED** [review]

**Pull request merged**